### PR TITLE
tgui-next cryo cells

### PR DESCRIPTION
**Pull request by spookydonut**
## About The Pull Request
This is my first try

## Why It's Good For The Game
faster i guess?

## Changelog
:cl:
add: cryocell ui converted to tgui-next
/:cl:

OLD
image

NEW
image

### CODE REVIEW

**Comment by Time-Green** [comment]
Pictures?

File: tgui-next/packages/tgui/interfaces/Cryo.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+            content="Eject" />
+        )}>
+        <LabeledList>
+          <LabeledList.Item label="Contents">
+            {data.isBeakerLoaded ? (
+              data.beakerContents ? (
+                <LabeledList>
+                  {data.beakerContents.map(beakerContent => (
+                    <LabeledList.Item
+                      key={damageType.id} >
```
Comment (willox) [inline tgui-next/packages/tgui/interfaces/Cryo.js]: The key here shouldn't be related to the damageType (also damageType does not exist in this scope.)

File: tgui-next/packages/tgui/interfaces/Cryo.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+            <Button
+              icon={data.autoEject ? "sign-out-alt" : "sign-in-alt"}
+              onClick={() => act(ref, 'autoeject')}
+              content={data.autoEject ? "Auto" : "Manual"} />
+          </LabeledList.Item>
+        </LabeledList>
+      </Section>
+      <Section
+        title="Beaker"
+        button={(
```
Comment (willox) [inline tgui-next/packages/tgui/interfaces/Cryo.js]: `button` property on this section should be `buttons`. The eject button doesn't appear without this change.

File: tgui-next/packages/tgui/interfaces/Cryo.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+            content={(
+              <Button
+                icon={data.isOperating ? "power-off" : "times"}
+                disabled={data.isOpen}
+                onClick={() => act(ref, 'power')}
+                content={data.isOperating ? "On" : "Off"} />
+            )} />
+          <LabeledList.Item
+            label="Temperature"
+            content={data.cellTemperature} K />
```
Comment (willox) [inline tgui-next/packages/tgui/interfaces/Cryo.js]: The K here is dangling. Could be solved with something like `content={data.cellTemperature + ' K'}` or look at how Canister.js does it for canister pressure.

**Comment by willox** [comment]
I noticed the screenshot in your PR description is not of a tgui-next ui.

It looks like this for me:
image

**Comment by spookydonut** [comment]
> 
> 
> I noticed the screenshot in your PR description is not of a tgui-next ui.
> 
> It looks like this for me:
> image

yeah thats the old one, i'm trying to get the new one working for me so i can post the new one alongside it

**Comment by spookydonut** [comment]
Done

File: tgui-next/packages/tgui/interfaces/Cryo.js [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+              data.beakerContents ? (
+                  data.beakerContents.map(beakerContent => (
+                    <Box
+                      key={beakerContent.id}
+                      color="pale-blue" >
+                      {beakerContent.volume} units of {beakerContent.name}
+                    </Box>
+                  ))
+              ) : (
+                "Beaker Empty"
```
Comment (willox) [inline tgui-next/packages/tgui/interfaces/Cryo.js]: This "Beaker Empty" branch won't be hit if beakerContents is an empty array.

**Review by stylemistake — APPROVED** [review]

**Comment by spookydonut** [comment]
Waiting on #47282 to be merged so I can build a new bundle

**Comment by MCterra10** [comment]
DO A BARREL ROLL!

File: tgui-next/packages/tgui/interfaces/Cryo.js [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+    {
+      label: "Burn",
+      type: "fireLoss",
+    },
+  ];
+
+  return (
+    <Fragment>
+      <Section
+        title="Occupant">
```
Comment (actioninja) [inline tgui-next/packages/tgui/interfaces/Cryo.js]: put on same line for single prop

File: tgui-next/packages/tgui/interfaces/Cryo.js [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+              <LabeledList.Item
+                label="State"
+                content={data.occupant.stat}
+                color={data.occupant.statstate} />
+              <LabeledList.Item
+                label="Temperature"
+                color={data.occupant.temperaturestatus}>
+                <AnimatedNumber value={data.occupant.bodyTemperature} /> K
+              </LabeledList.Item>
+              <LabeledList.Item
```
Comment (actioninja) [inline tgui-next/packages/tgui/interfaces/Cryo.js]: Same

File: tgui-next/packages/tgui/interfaces/Cryo.js [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+                    <AnimatedNumber value={data.occupant[damageType.type]} />
+                  </ProgressBar>
+                </LabeledList.Item>
+              )))}
+            </Fragment>
+          )}
+        </LabeledList>
+      </Section>
+      <Section
+        title="Cell">
```
Comment (actioninja) [inline tgui-next/packages/tgui/interfaces/Cryo.js]: same

File: tgui-next/packages/tgui/interfaces/Cryo.js [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+            content={(
+              <Button
+                icon={data.isOperating ? "power-off" : "times"}
+                disabled={data.isOpen}
+                onClick={() => act(ref, 'power')}
+                content={data.isOperating ? "On" : "Off"}
+                color={data.isOperating && ("green")} />
+            )} />
+          <LabeledList.Item
+            label="Temperature">
```
Comment (actioninja) [inline tgui-next/packages/tgui/interfaces/Cryo.js]: same

File: tgui/src/interfaces/cryo.ract [hunk 9]
```diff
@@ -1,53 +0,0 @@
-<ui-display title='Occupant'>
```
Comment (actioninja) [inline tgui/src/interfaces/cryo.ract]: don't delete the old ract file, we're keeping them around until old tgui is deprecated for reference and backup purposes.

File: tgui-next/packages/tgui/interfaces/Cryo.js [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+          <LabeledList.Item
+            label="Power"
+            content={(
+              <Button
+                icon={data.isOperating ? "power-off" : "times"}
+                disabled={data.isOpen}
+                onClick={() => act(ref, 'power')}
+                content={data.isOperating ? "On" : "Off"}
+                color={data.isOperating && ("green")} />
+            )} />
```
Comment (actioninja) [inline tgui-next/packages/tgui/interfaces/Cryo.js]: put this in the tag instead of using `content`, easier to follow

**Review by actioninja — CHANGES_REQUESTED** [review]
minor style changes

File: tgui-next/packages/tgui/interfaces/Cryo.js (hunk 10 above)
Comment (stylemistake) [reply tgui-next/packages/tgui/interfaces/Cryo.js]: indeed

**Review by stylemistake — APPROVED** [review]

**Pull request merged**